tools/azsdk-cli/docs/specs/0-scenario-2.spec.md

```diff
@@ -0,0 +1,818 @@
+- Azure subscription permits on-demand resource provisioning and teardown
+
+### In Scope for Scenario 2
+
+- **All [Scenario 1 activities](./0-scenario-1.spec.md#workflow) are included**
+- All five languages: .NET, Java, JavaScript, Python, Go
+- **Both [Net-New SDKs](#net-new-sdk) and existing SDKs**
+- TypeSpec-based generation for the Health Deidentification service
+- Both **data plane and management plane** coverage
+- Both **Agent Mode and CLI Mode** validation
+- VS Code with GitHub Copilot guidance and automation
+- AI models: Claude Sonnet 4 / 4.1, GPT-4, GPT-5
+- Automated **[Environment Remediation](#environment-remediation)** (install/upgrade tooling)
+- **[TypeSpec Customizations](#typespec-customizations)** to shape generated surfaces
+- **[Code Customizations](#code-customizations)** to add handwritten layers
+- Creation of **[Live Tests](#live-test)**, **[Test Assets](#test-asset)**, and new **[Test Recordings](#test-recording)** for [Net-New SDKs](#net-new-sdk)
+- Re-recording flows for existing SDKs
+- Automated teardown of Azure resources after live runs
+
+### Out of Scope for Scenario 2
+
+- Outer-loop activities such as committing changes, creating PRs, release management, or package publishing
+- GA releases
+- Architectural review preparation
+- Breaking change reviews
+- Updating changelog/`README.md`/metadata content for data-plane libraries
+- Error resolution assistance beyond environment setup remediation
+- Platform support beyond Windows
+- Editors other than VS Code
+
+---
+
+## Workflow
+
+**Legend**: 🆕 = New tool for Scenario 2 | Existing tool from Scenario 1
+
+1. **Environment Setup** → `azsdk_verify_setup`, 🆕 `azsdk_package_setup_env`
+   - Verify tools and versions (see [Scenario 1 – Environment Setup](./0-scenario-1.spec.md#1-environment-setup))
+   - Remediate missing or out-of-date tools with azsdk_package_setup_env
+
+2. **Generating** → `azsdk_package_generate_code`
+   - Generate SDK code, tests, and samples (see [Scenario 1 – Generating](./0-scenario-1.spec.md#2-generating))
+   - **Note**: Generation tooling now handles library project bootstrapping for [Net-New SDKs](#net-new-sdk)
+
+3. **Determine Customization Approach** → 🆕 `azsdk_package_customization_playbook`
+   - Consult playbook to determine whether customizations should be done via TypeSpec or code
+   - Get decision criteria and rationale based on Azure SDK guidelines
+
+4. **TypeSpec Customizations** → 🆕 `azsdk_package_customize_typespec`
+   - Modify client.tsp
+   - Validate TypeSpec correctness
+
+5. **Code Customizations** → 🆕 `azsdk_package_customized_code_update`
+   - Add handwritten/custom code on top of generated output
+
+6. **Testing** → `azsdk_package_run_tests`, 🆕 `azsdk_package_test_resources_manage`, 🆕 `azsdk_package_test_assets_create`, 🆕 `azsdk_package_test_mode_set`
+
+   If net-new SDK:
+   - Create test asset files with `azsdk_package_test_assets_create`
```

Comment:
I feel like this should all be within one or two tools and the level of granularity shouldnt be published to the user but rather determined internally by use based on the state of the package and context of the discussion

Reply:
Makes sense to me I'll update the scenario to follow this guidance.